### PR TITLE
Fix query lookup when using Money object

### PR DESCRIPTION
**Pull request by VictorTan93**
### Fix query lookup when using Money object

This PR fixes an issue with Django ORM query lookups when using `Money` objects directly in filters.

Previously, querying with a `Money` object (e.g. `Model.objects.filter(price=Money(100, 'USD'))`) would only compare the amount without checking the currency, leading to incorrect results when multiple currencies are present in the database.

---

### 📝 Changes

Core Fix:

Implemented custom lookup classes (`MoneyExactLookup`, `MoneyLtLookup`, `MoneyLteLookup`, `MoneyGtLookup`, `MoneyGteLookup`):
- Properly handle `Money` objects by checking both amount and currency in the SQL query
- Reference: https://docs.djangoproject.com/en/4.0/howto/custom-lookups/

Moving away from `MoneyManager` approach, because it is easily overwritten when a custom manager is used in a Django Model (i.e. `objects = CustomManager()`).

Code Cleanup:

- Removed `MoneyManager` class
- Extracted `MoneyFieldProxy` into its own module (`proxy.py`) for better organization
- Extracted utility functions (`currency_field_name`, `currency_field_db_column`) into `utils.py`

Testing:

Added comprehensive test suite in `test_django_lookups.py` covering:
- Exact lookups with Money objects
- Greater than/less than comparisons
- Greater than or equal to / less than or equal to comparisons
- Edge cases and currency validation

---

### Behavior

When filtering with a `Money` object:
```python
Product.objects.filter(price=Money(100, 'USD'))
```

The generated SQL now checks both the amount column AND the currency column:
```sql
WHERE (price = 100 AND price_currency = 'USD')
```

This ensures that only records with matching currency are returned, preventing incorrect comparisons between amounts in different currencies.

---

### ✅ Pre-review checklist for the PR author

- [x] I have tested my code locally on my machine
- [ ] I have written tests (if appropriate)
- [x] I have added appropriate label to PR

---

### CODE REVIEW

File: money/tests/test_django.py [hunk 1]
```diff
@@ -115,95 +102,6 @@ def test_defaults_as_separate_values(self) -> None:
         ent = MoneyModelDefaults.objects.get(pk=ent.id)
         assert ent.price == Money(100, "USD")
 
-    def test_lookup(self) -> None:
```
Comment (VictorTan93) [inline money/tests/test_django.py:118]: Moved to `test_django_lookups.py`

File: money/contrib/django/models/fields.py [hunk 2]
```diff
@@ -141,6 +74,8 @@ def value_to_string(self, obj):
 
 class MoneyField(InfiniteDecimalField):
     description = gettext_lazy("An amount and type of currency")
+
+    add_currency_field: bool
```
Comment (ewendlick) [inline money/contrib/django/models/fields.py:78]: NAB: Very minor, but what do you think about calling this something a little different, like `contains_currency_field`, `has_currency_field`, `with_currency_field`, or `using_currency_field`?

"add_currency_field" feels a little ambiguous to me, whether it is for adding a currency field to an existing instance, for positive amounts, or if it is a method

Comment (VictorTan93) [reply money/contrib/django/models/fields.py:78]: Ahhh this is not introduced by me, this line was added for `mypy` typing. 
Changing this will affect everyone using it

**Review by ewendlick — APPROVED** [review]
